DEVELOPMENT.md

```diff
@@ -16,8 +16,51 @@ With this repository cloned locally, execute the following commands in a termina
+
+**Adding the necessary files for a new Music Provider**
+
+Add a new folder to the `providers` folder with the name of provider. Add two files inside:
+1. `__init__.py`. This file contains the Python code of your provider.
+2. `manifest.json`. This file contains metadata and configuration for your provider.
+
+**Configuring the manifest.json file**
+
+The easiest way to get start is to copy the contents of the manifest of an existing Music Provider, e.g. Spotify or Youtube Music. The manifest can contain the following properties:
+| Name  | Description  | Type  |
+|---|---|---|
+| type  | Always `music`  | string  |
+| domain  | The internal unique id of the provider, e.g. `spotify` or `ytmusic`  | string  |
+| name  | The full name of the provider, e.g. `Spotify` or `Youtube Music`  | string  |
+| description  | The full description of the provider  | string  |
+| codeowners  | List of Github names of the codeowners of the provider  | array[string]  |
+| config_entries  | List of configurable properties for the provider, e.g. `username` or `password`*. | array[object]  |
+| config_entries.key  | The unique key of the config entry, used to obtain the value in the provider code  | string  |
+| config_entries.type  | The type of the config entry. Possible values: `string`, `secure_string` (for passwords), `boolean`, `float`, `integer`, `label` (for a single line of text in the settings page)  | string  |
+| config_entries.label | The label of the config entry. Used in the settings page | string |
+| requirements | List of requirements for the provider in pip string format. Supported values are `package==version` and `git+https://gitrepoforpackage` | array[string]
+| documentation | URL to the Github discussion containing the documentation for the provider. | string |
+| multi_instances | Whether multiple instances of the configuration are supported, e.g. multiple user accounts for Spotify | boolean |
+
+\* These `config_entries` are used to automatically generate the settings page for the provider in the front-end. The values can be obtained via `self.config.get_value(key)`.
+
+**Creating the provider**
+
+Create a file called `__init__.py` inside the folder of your provider. This file will contain the logic for the provider. All Music Providers must inherit from the [`MusicProvider`](./music_assistant/server/models/music_provider.py) base class and override the necessary functions where applicable. A few things to note:
+* The `setup()` function is called by Music Assistant upon initialization of the provider. It gives you the opportunity the prepare the provider for usage. For example, logging in a user or obtaining a token can be done in this function.
+* A provider should let Music Assistant know which [`ProviderFeatures`](./music_assistant/common/models/enums.py) it supports by implementing the property `supported_features`, which returns a list of `ProviderFeatures`.
+* Music Assistant support multiple ways to playback music:
```

Comment:
I'd explain the two step procedure here with getting the stream details first and playback later